cli/azd/internal/repository/infra_confirm.go

```diff
@@ -103,10 +102,43 @@ func (i *Initializer) infraSpecFromDetect(
 		if svc.Docker == nil || svc.Docker.Path == "" {
 			// default builder always specifies port 80
 			serviceSpec.Port = 80
-
 			if svc.Language == appdetect.Java {
 				serviceSpec.Port = 8080
 			}
+		} else {
+			ports := svc.Docker.Ports
+			if len(ports) == 0 {
+				port, err := i.getPortByPrompt(ctx, serviceSpec.Name)
```

Comment:
Let's keep the experience the same for this initial prompt, i.e. the prompt message should be:

"What port does '" + serviceSpec.Name + "' listen on?

Reply:
For the other flow, where we did detect multiple ports, what you have looks good.
In that flow, it looks like:

> $ What port does 'api' listen on? Other
> $ Provide the port number for 'api': 80

which looks perfectly fine!

Reply:
Updated.